tendermint/tests/lite.rs

```diff
@@ -37,6 +39,86 @@ struct LiteBlock {
     next_validator_set: Set,
 }
 
+#[derive(Deserialize, Clone, Debug)]
+struct TestBisection {
+    description: String,
+    trust_options: TrustOptions,
+    primary: Provider,
+    height_to_verify: Height,
+    // TODO: make trust_level deserialize from string instead of integer
+    // TODO: trust_level should go under TrustOptions
+    trust_level: TrustThresholdFraction,
+    now: Time,
+    expected_output: Option<String>,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+struct Provider {
+    chain_id: String,
+    lite_blocks: Vec<LiteBlock>,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+struct TrustOptions {
+    // TODO: add trust_level here
+    period: Duration,
+    height: Height,
+    hash: Hash,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+struct MockRequester {
+    chain_id: String,
+    signed_headers: HashMap<u64, SignedHeader>,
+    validators: HashMap<u64, Set>,
+}
+
+type LightSignedHeader = lite::types::SignedHeader<SignedHeader, Header>;
+
+impl Requester<SignedHeader, Header> for MockRequester {
+    fn signed_header(&self, h: u64) -> Result<LightSignedHeader, Error> {
+        println!("requested signed header for height:{:?}", h);
+        if let Some(sh) = self.signed_headers.get(&h) {
+            return Ok(sh.into());
+        }
+        println!("couldn't get sh for: {}", &h);
+        Err(Error::RequestFailed)
+    }
+
+    fn validator_set(&self, h: u64) -> Result<Set, Error> {
+        println!("requested validators for height:{:?}", h);
+        if let Some(vs) = self.validators.get(&h) {
+            return Ok(vs.to_owned());
+        }
+        println!("couldn't get vals for: {}", &h);
+        Err(Error::RequestFailed)
+    }
+}
+
+impl MockRequester {
+    fn new(chain_id: String, lite_blocks: Vec<LiteBlock>) -> Self {
+        let mut sh_map: HashMap<u64, SignedHeader> = HashMap::new();
+        let mut val_map: HashMap<u64, Set> = HashMap::new();
+        let last_block = lite_blocks
+            .get(lite_blocks.len() - 1)
```

Comment:
TODO: clippy fails here, suggests to use lite_block.last() instead. But then fails to compile

Reply:
I'll look into it now.

Reply:
Huh, this actually works. You can apply the suggestion here, pull locally, apply `cargo fmt --` and we are good.
```suggestion
            .last()
```